### DISCUSSION TITLE
Traffic mirroring/shadowing for Kong Gateway or any alternatives?

### DISCUSSION THREAD

**Discussion by nhp0712** [Kong/kong#7496] [Help] [answered]
Hi all, 

I'm eagerly looking for a plugin or any kind of third party to test the traffic mirroring/shadowing for Kong Gateway. I've found this similar question in Kong community [https://discuss.konghq.com/t/kong-for-shadowing-or-mirroring-production-traffic/2010/6](url)... And it looks like traffic mirroring plugin for Kong is under implementation process.

Can I get any updates of its progress? And is there any other way to test it on Kong at the moment?

Thanks a lot!

**Comment by dndx** [comment, marked as answer]
Hello,

Kong does not currently have such functionality, however you can use custom `nginx.conf` templates and enable the http://nginx.org/en/docs/http/ngx_http_mirror_module.html module on Kong's `server` block to mirror the request somewhere else.

**Comment by anuj147** [comment]
Hey guys,

Since we also have this requirement, we are planning to build a plugin that can shadow traffic to an alternate service without waiting for a response from them. Something on the lines of what other gateways such as Ambassador and KrakenD provide.
https://www.getambassador.io/docs/edge-stack/latest/topics/using/shadowing/
https://www.krakend.io/docs/backends/shadow-backends/

I think this plugin should be straightforward to do implement this using ngx-http module and can be applied on a route/service/consumer.
